apps/condo/domains/billing/schema/BillingReceiptFile.js

```diff
@@ -2,30 +2,77 @@
  * Generated by `createschema billing.BillingReceiptFile 'file:File;context:Relationship:BillingIntegrationOrganizationContext:CASCADE;receipt:Relationship:BillingReceipt:CASCADE;controlSum:Text'`
  */
 
-const { Text, Relationship, File } = require('@keystonejs/fields')
+const { Text, Relationship, File, Virtual } = require('@keystonejs/fields')
+const get = require('lodash/get')
 
 const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
-const { GQLListSchema, getById } = require('@open-condo/keystone/schema')
+const { GQLListSchema, getById, find } = require('@open-condo/keystone/schema')
 
 const access = require('@condo/domains/billing/access/BillingReceiptFile')
 const { UNEQUAL_CONTEXT_ERROR } = require('@condo/domains/common/constants/errors')
 const FileAdapter = require('@condo/domains/common/utils/fileAdapter')
-const { getFileMetaAfterChange } = require('@condo/domains/common/utils/fileAdapter')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
 
 const BILLING_RECEIPT_FILE_FOLDER_NAME = 'billing-receipt-pdf'
 const Adapter = new FileAdapter(BILLING_RECEIPT_FILE_FOLDER_NAME)
-const fileMetaAfterChange = getFileMetaAfterChange(Adapter, 'file')
 
+const isResidentVerified = async ({ id, phone }, billingAccountId) => {
+    const { unitName, unitType, property } = await getById('BillingAccount', billingAccountId)
+    const billingProperty = await getById('BillingProperty', property)
+    const { address } = billingProperty // TODO: Use addressKey, in tests addressKey has different values for the same address
+    const contacts = await find('Contact', {
+        phone,
+        isVerified: true,
+        unitName, unitType,
+        property: { address },
+    })
+    return contacts.length !== 0
+}
 
 const BillingReceiptFile = new GQLListSchema('BillingReceiptFile', {
     schemaDoc: 'File for billing receipt',
     fields: {
 
         file: {
-            schemaDoc: 'File object with meta information about the receipt',
+            schemaDoc: 'Wrapper to return file version of the receipt (usually PDF) with personal information or without',
+            type: Virtual,
+            graphQLReturnType: 'File',
+            resolver: async (item, _, { authedItem }) => {
+                // We are changing link to publicData only for not verified residents. In other cases we return sensitive data files
+                let file = item.sensitiveDataFile
+                if (authedItem.type === RESIDENT) {
+                    // Resident already has an access to billing receipt, so we only need to check if he is approved
+                    const { account } = await getById('BillingReceipt', item.receipt)
+                    const isApproved = await isResidentVerified(authedItem, account)
+                    if (!isApproved) {
+                        file = item.publicDataFile
```

Comment:
It's better not to use sensitive data as default value
```suggestion
                let file = item.publicDataFile
                if (authedItem.type === RESIDENT) {
                    // Resident already has an access to billing receipt, so we only need to check if he is approved
                    const { account } = await getById('BillingReceipt', item.receipt)
                    const isApproved = await isResidentVerified(authedItem, account)
                    if (isApproved) {
                        file = item.sensitiveDataFile
```

Reply:
Done